### ISSUE TITLE
cmnPrintf.h  no viable conversion from returned value of type 'std::ostream'  on mac

### ISSUE THREAD

**Issue by marcinbalicki** [jhu-cisst/cisst#46]
```
In file included from ~/dev/src/cisst-saw/sawATIForceSensor/examples/mainSimulator.cpp:23:
In file included from ~/dev/build/cisst-saw-x/cisst/include/cisstCommon.h:45:
~/dev/src/cisst-saw/cisst/cisstCommon/cmnPrintf.h:315:16: error: no viable conversion from returned value of type 'std::ostream' (aka 'basic_ostream<char>') to function return type 'bool'
        return OutputStream;
               ^~~~~~~~~~~~
```

The cmake macro  CISST_OSTREAM_CAN_CAST_TO_INT to 1

Setup:
latest devel branch on Mac OS Sierra

Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.5.0

_Not sure why this is happening but here is a workaround:_

** One fix:**
```
remove : #include <cisstCommon.h> from sawATIForceSensor/examples/mainSimulator.cpp:23
change #include <cisstCommon.h> to #include <cisstCommon/cmnUnits.h> in sawATIForceSensor/sawATINetFTSimulatorQtWidget.cpp:23
```

**Comment by adeguet1** [comment]
The root cause is clearly in cmnPrintf.  I must admit I haven't compiled on Mac OS for a while and I'm still running El Capitan so I might not be able to reproduce the issue.

Feel free to clone sawATIForceSensor and issue a pull request, your workaround is good and I'll approve.

On the long run, I might mark all the library wide header files (cisstCommon.h, cisstVector.h, ...) as deprecated to make sure developers don't use them.

**Comment by marcinbalicki** [comment]
Updated sawATIForceSensor. @adeguet1 - can you please update the cisst-saw module link to point to the latest master branch commit. Thanks.

**Comment by adeguet1** [comment]
I've updated the devel and master branches of cisst-saw to point to the latest repositories.

**Comment by adeguet1** [comment]
This has been fixed in the latest cisst devel branch, see f491d3027f35391654236ff649fec24ae7ebc58f
